src/datatable/sphinxext/xfunction.py

```diff
@@ -1112,6 +1096,183 @@ def _generate_body(self):
 
 
 
+#-------------------------------------------------------------------------------
+# Helper functions
+#-------------------------------------------------------------------------------
+
+def locate_python_variable(name, lines):
+    """
+    Find declaration of a variable `name` within python source `lines`.
+
+    Returns a tuple (istart, iend) such that `lines[istart:iend]` would
+    contain the lines where the variable is declared. Variable declaration
+    must have the form of an assignment, i.e. "{name} = ...". Other ways
+    of declaring a variable are not supported.
+
+    A ValueError is raised if variable declaration cannot be found.
+
+    NYI: detection of multi-line variable declarations, such as dicts,
+    lists, multi-line strings, etc.
+    """
+    assert isinstance(name, str)
+    rx = re.compile(r"\s*%s\s*=\s*\S" % name)
+    for i, line in enumerate(lines.data):
+        if re.match(rx, line):
+            return (i, i+1)
+    raise ValueError("Could not find variable `%s` in <FILE>" % name)
+
+
+
+def locate_python_function(name, kind, lines):
+    """
+    Find declaration of a class/function `name` in python source `lines`.
+
+    Returns a tuple (istart, iend) such that `lines[istart:iend]` would
+    contain the lines where the class/function is declared. The body of
+    the class/function starts with "(class|def) {name}..." and ends
+    at a line with the indentation level equal or smaller than the
+    level of the declaration start.
+
+    A ValueError is raised if variable declaration cannot be found.
```

Comment:
Back-ticks around "ValueError".

Reply:
This is an internal (not user-visible) documentation so I only use backticks to disambiguate code from regular text. For example, "source lines" is just a description, whereas "source \`lines\`" specifically refers to the variable `lines`

Reply:
Yes, but at some point the internal docs may go public fully or partially, like in the case of the new options, etc.